samples/drivers/dac/src/main.c

```diff
@@ -8,48 +8,32 @@
 #include <sys/printk.h>
 #include <drivers/dac.h>
 
-#if defined(CONFIG_BOARD_NUCLEO_F091RC) || \
-	defined(CONFIG_BOARD_NUCLEO_G071RB) || \
-	defined(CONFIG_BOARD_NUCLEO_G431RB) || \
-	defined(CONFIG_BOARD_NUCLEO_L073RZ) || \
-	defined(CONFIG_BOARD_NUCLEO_L152RE) || \
-	defined(CONFIG_BOARD_NUCLEO_F429ZI) || \
-	defined(CONFIG_BOARD_NUCLEO_F767ZI) || \
-	defined(CONFIG_BOARD_RONOTH_LODEV)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac1))
-#define DAC_CHANNEL_ID		1
-#define DAC_RESOLUTION		12
-#elif defined(CONFIG_BOARD_TWR_KE18F)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac0))
-#define DAC_CHANNEL_ID		0
-#define DAC_RESOLUTION		12
-#elif defined(CONFIG_BOARD_FRDM_K64F)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac0))
-#define DAC_CHANNEL_ID		0
-#define DAC_RESOLUTION		12
-#elif defined(CONFIG_BOARD_FRDM_K22F)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac0))
-#define DAC_CHANNEL_ID		0
-#define DAC_RESOLUTION		12
-#elif defined(CONFIG_BOARD_ARDUINO_ZERO)
-#define DAC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(dac0))
-#define DAC_CHANNEL_ID		0
-#define DAC_RESOLUTION		10
+#define ZEPHYR_USER_NODE DT_PATH(zephyr_user)
+
+#if (DT_NODE_HAS_PROP(ZEPHYR_USER_NODE, dac) && \
+	DT_NODE_HAS_PROP(ZEPHYR_USER_NODE, dac_channel_id) && \
+	DT_NODE_HAS_PROP(ZEPHYR_USER_NODE, dac_resolution))
+#define DAC_NODE DT_PHANDLE(ZEPHYR_USER_NODE, dac)
+#define DAC_CHANNEL_ID DT_PROP(ZEPHYR_USER_NODE, dac_channel_id)
+#define DAC_RESOLUTION DT_PROP(ZEPHYR_USER_NODE, dac_resolution)
 #else
-#error "Unsupported board."
+#error "Unsupported board: see README and check /zephyr,user node"
+#define DAC_NODE DT_INVALID_NODE
+#define DAC_CHANNEL_ID 0
+#define DAC_RESOLUTION 0
```

Comment:
are these definitions needed?

Reply:
Without them, you'll get build errors later down in the file where they are missing. With them, you just get the `#error` on line 20. This is more readable error output, which is important for a sample.